tests/test_controller.py

```diff
@@ -2572,6 +2573,48 @@ def test_device_id_handler_bad_method(self):
             eq_("Only DELETE is supported.", response.detail)
 
 
+class TestODLNotificationController(ControllerTest):
+    """Test that an ODL distributor can notify the circulation manager
+    when a loan's status changes."""
+
+    def test_notify_success(self):
+        collection = MockODLWithConsolidatedCopiesAPI.mock_collection(self._db)
+        patron = self._patron()
+        pool = self._licensepool(None, collection=collection)
+        pool.licenses_owned = 10
+        pool.licenses_available = 5
+        loan, ignore = pool.loan_to(patron)
+        loan.external_identifier = self._str
+
+        with self.request_context_with_library("/", method="POST"):
+            flask.request.data = json.dumps({
+               "id": loan.external_identifier,
+               "status": "revoked",
+            })
+            response = self.manager.odl_notification_controller.notify(
+                loan.external_identifier)
+            eq_(200, response.status_code)
+
+            # The pool's availability has been updated.
+            api = self.manager.circulation_apis[self._default_library.id].api_for_license_pool(loan.license_pool)
+            eq_([loan.license_pool], api.availability_updated_for)
```

Comment:
It should also check the effect on the Loan object, right?

Reply:
`update_loan` is tested separately, but I don't think you have a test that `notify` calls `update_loan`, and this would be the place to put it.

Reply:
I implemented `update_loan` in the mock api class and had it set `api.availability_updated_for`, so this line verifies that it was called.

Reply:
Ok, cool.